v2/client.go

```diff
@@ -53,11 +53,14 @@ func NewClient(config *ClientConfiguration) (Client, error) {
 	c.doRequestFunc = c.doRequest
 
 	if config.AuthConfig != nil {
-		if config.AuthConfig.BasicAuthConfig == nil {
-			return nil, errors.New("BasicAuthConfig is required if AuthConfig is provided")
+		if config.AuthConfig.BasicAuthConfig == nil && config.AuthConfig.BearerConfig == nil {
+			return nil, errors.New("Non-nil AuthConfig cannot be empty")
```

Comment:
Maybe drop the non-nil, so either just "AuthConfig..." or maybe "Existing AuthConfig...". Users probably don't necessarily understand go lingo :)